test/ClangImporter/cxx_interop.swift

```diff
@@ -9,3 +9,9 @@ do {
   tmp.a = 3
   tmp.b = nil
 }
+
+// UsingShadowDecl importing
+do {
+  // In C++: using ns::NamespacedType;
+  let _ = NamespacedType.self
```

Comment:
I think like #26339 we should have a mangling test here, to make sure the importer is in fact using a typealias for something namespaced rather than moving the declaration to the top level.